tests/fast-integration/eventing-test/eventing-test.js

```diff
@@ -473,6 +486,130 @@ describe('Eventing tests', function() {
+        });
+
+        it('Delete the eventing-upgrade-sink', async function() {
+          await undeployEventingFunction(eventingUpgradeSinkName);
+          debug(`checking if eventing upgrade sink is not alive anymore...`);
+          await checkFunctionUnreachable(eventingUpgradeSinkName, testNamespace, clusterHost);
+        });
+
+        it('Generate new eventId, save the id and publish events', async function() {
+          let existingCMData = {};
+          // get existing configMap
+          const cm = await getConfigMapWithRetries(testDataConfigMapName, testNamespace);
+          if (cm && cm.data) {
+            existingCMData = cm.data;
+          }
+
+          const eventIdBinary = getRandomEventId(encoding);
+          await createK8sConfigMap(
+              {
+                ...existingCMData,
+                upgradeEventID: eventIdBinary,
+              },
+              testDataConfigMapName,
+          );
+          // publish the event, which should be delivered after the upgrade
+          await publishEventWithRetry(clusterHost, encoding, eventIdBinary,
+              subscriptionsTypes[0].type, subscriptionsTypes[0].source);
+        });
+      });
+
+      context('Post-upgrade tasks to verify that event is delivered after sink is available', function() {
+        let eventID;
+        before('Check if this is pre-upgrade stage', async function() {
+          if (upgradeStage !== 'post') {
+            debug(`Skipping post-upgrade tasks...`);
+            this.skip();
+          }
+
+          const cm = await getConfigMapWithRetries(testDataConfigMapName, 'default');
+          if (!cm || !cm.data || !cm.data.upgradeEventID) {
+            debug(`Skipping post-upgrade tasks because config map is not configured...`);
+            this.skip();
+          }
+          eventID = cm.data.upgradeEventID;
+        });
+
+        it(`deploy again and wait for function: ${eventingUpgradeSinkName}`, async function() {
+          await deployEventingSinkFunction(eventingUpgradeSinkName);
+          await waitForEventingSinkFunction(eventingUpgradeSinkName);
+          debug(`checking if eventing upgrade sink is reachable through the api rule`);
+          await checkFunctionReachable(eventingUpgradeSinkName, testNamespace, clusterHost);
+        });
+
+        it('Label subscription to trigger reconciliation', async function() {
+          subscriptions[0].metadata.labels = {
+            'changed': 'now',
+          };
+          await k8sApply(subscriptions, testNamespace);
+          await waitForSubscription(subName, testNamespace, 'v1alpha2');
+        });
+
+        it('Wait for the pending event to be delivered', async function() {
+          await ensureEventReceivedWithRetry(eventingUpgradeSinkName, clusterHost,
+              encoding, eventID, subscriptionsTypes[0].type, subscriptionsTypes[0].source);
+        });
+
+        it(`Delete configMap: ${testDataConfigMapName}`, async function() {
```

Comment:
I woud have clean-up in `after` functions as it is run even if test fails. `it` function doesn't run for test failures.

Reply:
It is also cleaned up in `after` function [here](https://github.com/kyma-project/kyma/blob/e9df8e870eaa78b144ebe2027440620f26384174/tests/fast-integration/eventing-test/utils.js#L156). It is removed here once the required test is completed.

Reply:
do we still need this clean-up here then?